core/src/serde.rs

```diff
@@ -0,0 +1,39 @@
+use proc_macro2::Span;
+use syn::Ident;
+
+use crate::util::IdentString;
+
+impl<'de> serde::Deserialize<'de> for IdentString {
```

Comment:
Please add a doc comment somewhere that says this serializes as a string and expects a string.